test/e2e/density.go

```diff
@@ -201,22 +221,27 @@ var _ = framework.KubeDescribe("Density", func() {
 		}
 		itArg := testArg
 		It(name, func() {
-			podsPerNode := itArg.podsPerNode
-			totalPods = podsPerNode * nodeCount
-			RCName = "density" + strconv.Itoa(totalPods) + "-" + uuid
 			fileHndl, err := os.Create(fmt.Sprintf(framework.TestContext.OutputDir+"/%s/pod_states.csv", uuid))
 			framework.ExpectNoError(err)
 			defer fileHndl.Close()
-			config := framework.RCConfig{Client: c,
-				Image:                "gcr.io/google_containers/pause:2.0",
-				Name:                 RCName,
-				Namespace:            ns,
-				PollInterval:         itArg.interval,
-				PodStatusFile:        fileHndl,
-				Replicas:             totalPods,
-				CpuRequest:           nodeCpuCapacity / 100,
-				MemRequest:           nodeMemCapacity / 100,
-				MaxContainerFailures: &MaxContainerFailures,
+			podsPerNode := itArg.podsPerNode
+			totalPods = podsPerNode * nodeCount
+			RCConfigs := make([]framework.RCConfig, podsPerNode)
+			for i := 0; i < podsPerNode; i++ {
```

Comment:
Although I like the refactoring itself, I would like to not change the test itself in this PR. Can you still make a single RC with totalPods replicas instead of multiple RCs? After all other changes, switching it should be very easy anyway.